### PR TITLE
Introduce multiple client

### PR DESCRIPTION
**Pull request by ipdae**
- Update RPC interface for multiple clients https://github.com/planetarium/NineChronicles.RPC.Shared/pull/15
- Add new activated query for launcher https://github.com/planetarium/9c-launcher/pull/891

### CODE REVIEW

**Review by area363 — APPROVED** [review]

File: NineChronicles.Headless/BlockChainService.cs [hunk 1]
```diff
@@ -157,5 +162,19 @@ public UnaryResult<bool> ReportException(string code, string message)
...
+        public UnaryResult<bool> AddClient(byte[] addressByte)
+        {
+            var address = new Address(addressByte);
+            _publisher.AddClient(address).Wait();
+            return UnaryResult(true);
+        }
+
+        public UnaryResult<bool> RemoveClient(byte[] addressByte)
+        {
+            var address = new Address(addressByte);
```
Comment (moreal) [inline NineChronicles.Headless/BlockChainService.cs:175]: For consistency:

```suggestion
        public UnaryResult<bool> RemoveClient(byte[] addressBytes)
        {
            var address = new Address(addressBytes);
```

File: NineChronicles.Headless/BlockChainService.cs [hunk 2]
```diff
@@ -157,5 +162,19 @@ public UnaryResult<bool> ReportException(string code, string message)
 
             return UnaryResult(true);
         }
+
+        public UnaryResult<bool> AddClient(byte[] addressByte)
+        {
+            var address = new Address(addressByte);
```
Comment (moreal) [inline NineChronicles.Headless/BlockChainService.cs:168]: For consistency:

```suggestion
        public UnaryResult<bool> AddClient(byte[] addressBytes)
        {
            var address = new Address(addressBytes);
```

File: NineChronicles.Headless/GraphTypes/ActivationStatusMutation.cs [hunk 3]
```diff
@@ -64,6 +66,61 @@ public ActivationStatusMutation(NineChroniclesNodeService service)
...
+                        context.Errors.Add(new ExecutionError("The given key isn't in the correct format.", ae));
+                        return false;
+                    }
+                    catch (Exception e)
+                    {
+                        var msg = "Unexpected exception occurred during ActivatedAccountsMutation: {e}";
+                        context.Errors.Add(new ExecutionError(msg, e));
+                        return false;
+                    }
+
```
Comment (moreal) [inline NineChronicles.Headless/GraphTypes/ActivationStatusMutation.cs:123]: This pull request seems for multiple RPC client. Then it looks like the way to make multiple users able to activate without local headless. But IMO it seems too dangerous and too hard to say "give your private key" to users. I wonder other opinions cc @area363 @longfin

File: NineChronicles.Headless/GraphTypes/ActivationStatusQuery.cs [hunk 4]
```diff
@@ -68,6 +69,62 @@ public ActivationStatusQuery(StandaloneContext standaloneContext)
                     }
                 }
             );
+
+            Field<NonNullGraphType<BooleanGraphType>>(
+                name: "addressActivated",
+                arguments: new QueryArguments(
+                    new QueryArgument<NonNullGraphType<AddressType>>
+                    {
+                        Name = "addressHex"
```
Comment (moreal) [inline NineChronicles.Headless/GraphTypes/ActivationStatusQuery.cs:78]: This is not `StringGraphType`. How about using `address` directly?
```suggestion
                        Name = "address"
```

File: NineChronicles.Headless/GraphTypes/ActivationStatusQuery.cs [hunk 5]
```diff
@@ -68,6 +69,62 @@ public ActivationStatusQuery(StandaloneContext standaloneContext)
...
+                        var userAddress = context.GetArgument<Address>("addressHex");
+                        Address activatedAddress = userAddress.Derive(ActivationKey.DeriveKey);
+
+                        if (blockChain.GetState(activatedAddress) is Bencodex.Types.Boolean)
+                        {
+                            return true;
+                        }
+
+                        // Preserve previous check code due to migration period.
+                        // TODO: Remove this code after v100061+
```
Comment (moreal) [inline NineChronicles.Headless/GraphTypes/ActivationStatusQuery.cs:106]: This check seems for `ActivateAccount0` and `ActivateAccount`. I may not understand this line perfectfully but this line should be removed. I understood like that the two ways should be used all.

File: NineChronicles.Headless/GraphTypes/ActivationStatusMutation.cs (hunk 3 above)
Comment (ipdae) [reply NineChronicles.Headless/GraphTypes/ActivationStatusMutation.cs:123]: i agree. so delete this commit and make #676

**Review by moreal — APPROVED** [review]

**Pull request merged**